storage/innobase/dict/dict0dict.cc

```diff
@@ -7444,26 +7444,83 @@ dberr_t dict_get_dictionary_info_by_id(ulint dict_id, char **name,
   return err;
 }
 
-bool dict_detect_encryption(bool is_upgrade, space_id_t mysql_plugin_space) {
-  bool encrypt_mysql = false;
-  if (is_upgrade) {
-    if (mysql_plugin_space == SYSTEM_TABLE_SPACE) {
-      return (srv_sys_space.is_encrypted());
-    }
+static bool dict_should_mysql_ibd_be_created_encrypted_due_to_upgrade(
+    space_id_t mysql_plugin_space) {
+  if (mysql_plugin_space == SYSTEM_TABLE_SPACE) {
+    return (srv_sys_space.is_encrypted());
+  }
 
-    space_id_t space_id = fil_space_get_id_by_name("mysql/plugin");
-    ut_ad(space_id != SPACE_UNKNOWN);
-    ut_ad(space_id == mysql_plugin_space);
+  space_id_t space_id = fil_space_get_id_by_name("mysql/plugin");
+  ut_ad(space_id != SPACE_UNKNOWN);
+  ut_ad(space_id == mysql_plugin_space);
 
-    fil_space_t *space = fil_space_get(space_id);
-    ut_ad(space != nullptr);
+  fil_space_t *space = fil_space_get(space_id);
+  ut_ad(space != nullptr);
 
-    if (space == nullptr) {
-      return (false);
-    }
-    encrypt_mysql = FSP_FLAGS_GET_ENCRYPTION(space->flags);
+  if (space == nullptr) {
+    return false;
   }
+  return FSP_FLAGS_GET_ENCRYPTION(space->flags);
+}
+
+static bool dict_should_mysql_ibd_be_create_encrypted() {
+  return srv_encrypt_tables == SRV_ENCRYPT_TABLES_ON ||
+         srv_encrypt_tables == SRV_ENCRYPT_TABLES_FORCE;
+}
+
+static bool dict_should_mysql_ibd_be_opened_as_encrypted(const char *filepath,
```

Comment:
Not very intuitive contract: all other bool dict_should_... functions return the "should?" answer, and this one returns success or failure instead, with "should?" in the out arg. But I have no idea of a good fix

Reply:
I have changed that to return error by parameter and should be encrypted by return